### ISSUE TITLE
Validate parameters to the "mapppings" sections in project.json

### ISSUE THREAD

**Issue by halter73** [dotnet/cli#3082]
## Steps to reproduce

Map a single embedded resource (meaning the destination/key doesn't end in a '/') to a directory. Use any of the invalid source and destination "mapping" parameters demonstrated below.
## Expected  behavior

An error output from `dotnet build` indicating that the destination must end in a '/' to map to a directory. An error for any invalid or mismatching source or destination parameters.
## Actual behavior

`dotnet build` embeds the single file in the directory if there is a single file. If there are zero or more than one files, no resources will be embedded the built dll/exe.
### Examples

For the purposes of demonstration, assume all referenced files and directories exist and the directories contain what their names imply:

```
"embed": {
  "mappings": {
    "": {
      "include": [ "single_file_folder" ]
    }
  }
}
```

No resource included. No build error for empty destination.

---

```
"embed": {
  "mappings": {
    "/":  "single_file_folder"
  }
}
```

No resource included. No build error for referencing a source directory without a
trailing '/' even though this works as an include parameter.

---

```
"embed": {
  "mappings": {
    "/test/": {
      "include": [ "single_file_folder" ]
    }
  }
}
```

".test.single_file_folder.single_file.txt" resource included. No build error.

The leading '.' is kind of weird. It probably should be an error to start
mapping destination with a '/' (which seems to get converted to a '.') if it
isn't the only chracter.

---

This same pattern leads to a worse issue in a packOptions mapping:

```
"packOptions": {
  "files": {
    "mappings": {
      "/Views2/":  "/Views/"
    }
  }
},
```

With this mapping, a hidden directory is added to the root of the nupkg. If it's
unpacked with explorer, it disappears.

![hidden-directory](https://cloud.githubusercontent.com/assets/54385/15333044/dd79bc08-1c1c-11e6-9d65-c5792b238381.png)

![hidden-directory2](https://cloud.githubusercontent.com/assets/54385/15333045/dedb7d8e-1c1c-11e6-9192-e98692785686.png)

---

```
"embed": {
  "mappings": {
    "test": "single_file_folder/"
  }
}
```

The single_file.txt file inside of single_file_folder is included as "test.txt"
resource. No build error for referencing a directory instead of a single file.

---

```
"embed": {
  "mappings": {
    "test": "multi_file_folder/"
  }
}
```

No resource included. No build error for referencing a directory instead of a
single file.

---

```
"embed": {
  "mappings": {
    "test": "empty_folder/"
  }
}
```

No resource included. No build error for referencing a directory instead of a
single file.

---

```
"embed": {
  "mappings": {
    "test": {
      "include": [ "file", "file.txt" ],
      "exclude": "file.txt"
    }
  }
}
```

`file` resource included as "test". If "file" is excluded `file.txt` resource is
included as "test.txt". If the mapping key is "test.foo" instead of "test"
either resource will be included as "test.foo" exactly. If nothing is
excluded, no resource will be added since there are multiple matching files.

I think it should result in a build error if an include array is specified for a destination not ending in a '/'.
## Environment data

`dotnet --info` output:

.NET Command Line Tools (1.0.0-preview1-002702)

Product Information:
 Version:     1.0.0-preview1-002702
 Commit Sha:  6cde21225e

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.10586
 OS Platform: Windows
 RID:         win10-x64

**Comment by blackdwarf** [comment]
I've just tried to repro some of these cases on the Preivew 4 tools (MSBuild-based). I've created the project.json file, then migrated using `dotnet migrate` and ran build. After that, I've checked the resulting assembly using dotpeek and the resources were embedded.

In general, however, with the new MSBuild-based tooling, I would expect most problems with embedding resources would be either directed to dotnet/sdk or Microsoft/MSBuild repos. 

Would be good to get a repo with this repro to further test and validate that the behavior is what is expected in MSBuild-based tooling.
